JumpscaleCore/clients/ssh/SSHClientBase.py

```diff
@@ -45,6 +45,10 @@ def _init(self, **kwargs):
         self.executor = j.tools.executor.ssh_get(self)
         self._wireguard = None
         self._init3()
+        if self.sshkey_name not in j.core.myenv.sshagent.key_names:
```

Comment:
The ssh client can be used without providing the `sshkey_name` if there is an agent that has the keys already loaded, in that case I think you should just check that `self.sshkey_name` is provided in the first place

Reply:
and actually we should check the value first in all cases because it doesnt make sense to check if "" in keys.
@BolaNasr this part can just be
```
if self.sshkey_name and self.sshk_name not in j.core.myenv.sshagent.key_names:
    j.core.myenv.sshagent.start()
    self.sshkey_obj.load()
```